SalesforceFoundation/EDA
# Preferred Email/Phone Translation Issue Fix

# Critical Changes

# Changes
## Updates to the Preferred Phone Functionality
The enhanced Preferred Phone functionality released in update 1.84 can now be enabled in orgs that translate Catalan, Spanish, or French.
- If you are an existing EDA customer who wants to enable the feature in an org that translates any of these languages, go to EDA Settings | Accounts and Contacts and check the "Enable Enhanced Preferred Phone Functionality" checkbox.
- If you're installing in a new org that translates any of these languages, the enhanced Preferred Phone logic is enabled by default, so you don't need to change the new setting. However, we do recommend that you uncheck "Disable Preferred Phone enforcement" if you want to ensure that Contacts have a Preferred Phone value when more than one Phone field is specified on the Contact record.
- If you're translating in any other languages and want to use this feature, use the translation workbench to update Preferred Phone and Preferred Email picklist values to match the translated field labels.

**Note:** Existing customers might notice that some of EDA's unit tests fail after this upgrade. It's not required that EDA unit tests pass in customer orgs, but if you want them to pass, updating the Contact's Preferred Phone picklist values in your org to "Home Phone," "Mobile Phone," "Other Phone," and "Work Phone" should fix failures related to the upgrade.

# Issues Closed
Fixes #946 

# New Metadata

# Deleted Metadata

# Testing Notes

Testing Preferred Phone:

1. Make sure Enhanced Preferred Phone Functionality is checked : EDA Settings → Accounts & Contacts → Enable Enhanced Preferred Phone Functionality:
2. Make sure "CON_PreferredPhone_TDTM" trigger handler is active in "Trigger handlers" Tab.
3. Create a Contact with Preferred Phone as ‘Home’ and enter a value in the field ‘Home Phone’ and try saving the record. The record saves.
4. Change the language from user → User record →  Change Language to Spanish OR  
    click your avatar on top right → Settings → Language and Time zone → Español
5. Now create a contact with ‘Teléfono elegido’ as ‘Teléfono particular’  and enter a value for the field ‘Teléfono particular’  and try to save the record, the record should be saved.

Testing Preferred Email:

1.  Make sure "CON_PreferredEmail_TDTM" trigger handler is active in "Trigger handlers" Tab.
2.Create a contact with Preferred Email as "Work" and enter a value in the field "Work Email: and try saving the record. The record saves.
3.  Change the language from user → User record →  Change Language to Spanish OR  
    click your avatar on top right → Settings → Language and Time zone → Español
4. Create a Contact with "Correo electrónico elegido" as "Correo electrónco laboral" and enter a value in "Correo electrónico laboral" and try to save the record, the record should be saved.

Test Push Upgrade for Old Customers vs New Install (New Customers)
Test Two+ Phones ("Tell us which is preferred.")
Test Two Emails ("Tell us which is preferred.")
Test Run Cleanup

Note : For old customers, the picklist values must match the field label values (except for in English.) For new customers, the picklist values must match the field label values for all 4 languages (English, French, Spanish, Catalan.)
